src/IssueViz/Editor/SelectedIssueTagging/Buffer/SelectedIssueLocationTagger.cs

```diff
@@ -0,0 +1,86 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Tagging;
+using SonarLint.VisualStudio.IssueVisualization.Editor.LocationTagging;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+using SonarLint.VisualStudio.IssueVisualization.Selection;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Editor.SelectedIssueTagging.Buffer
+{
+    internal class SelectedIssueLocationTagger: FilteringTaggerBase<IIssueLocationTag, ISelectedIssueLocationTag>
+    {
+        private readonly IAnalysisIssueSelectionService issueSelectionService;
+
+        private bool disposedValue;
+
+        public SelectedIssueLocationTagger(ITagAggregator<IIssueLocationTag> tagAggregator, ITextBuffer textBuffer, IAnalysisIssueSelectionService issueSelectionService)
+            : base(tagAggregator, textBuffer)
+        {
+            this.issueSelectionService = issueSelectionService;
+
+            // Changing any of the selected issue/flow/location will always result in the
+            // "SelectedLocationChanged" event being raised
+            issueSelectionService.SelectionChanged += OnSelectionChanged;
+        }
+
+        private void OnSelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            if (e.SelectionChangeLevel == SelectionChangeLevel.Flow || e.SelectionChangeLevel == SelectionChangeLevel.Issue)
+            {
+                UpdateWholeSpan();
```

Comment:
I think this should be renamed to NotifyTagsChanged or something of the sort. 
It occurred to me when I was looking at the tests, where we raise an event with `null` issue/flow, and I was confused for a moment how did that not fail, since we check in Filter that it's not null. And I had to re-remind myself when looking at the code that  UpdateWholeSpan just raises an event, and it doesn't execute the GetTags implementation. Hence why I think it should be more explicit that it's just an event raise.